packages/ra-ui-materialui/src/input/TranslatableInputs.spec.tsx

```diff
@@ -0,0 +1,192 @@
+                    fr: 'french description',
+                },
+                nested: {
+                    field: {
+                        en: 'english nested field',
+                        fr: 'french nested field updated',
+                    },
+                },
+            },
+            undefined
+        );
+    });
+
+    it('should allow to customize the locale selector', () => {
+        const Selector = () => {
+            const {
+                locales,
+                selectLocale,
+                selectedLocale,
+            } = useTranslatableContext();
+
+            const handleChange = (event): void => {
+                selectLocale(event.target.value);
+            };
+
+            return (
+                <select
+                    aria-label="select locale"
+                    onChange={handleChange}
+                    value={selectedLocale}
+                >
+                    {locales.map(locale => (
+                        <option
+                            key={locale}
+                            value={locale}
+                            id={`translatable-header-${locale}`}
+                        >
+                            {locale}
+                        </option>
+                    ))}
+                </select>
+            );
+        };
+
+        const { getByLabelText, queryByDisplayValue } = renderWithRedux(
+            <FormWithRedirect
+                record={record}
+                render={({ handleSubmit }) => (
+                    <form onSubmit={handleSubmit}>
+                        <TranslatableInputs
+                            locales={['en', 'fr']}
+                            selector={<Selector />}
+                        >
+                            <TextInput source="name" />
+                            <TextInput source="description" />
+                            <TextInput source="nested.field" />
+                        </TranslatableInputs>
+                        <button type="submit">save</button>
+                    </form>
+                )}
+            />
+        );
+
+        expect(getByLabelText('en').getAttribute('hidden')).toBeNull();
+        expect(getByLabelText('fr').getAttribute('hidden')).not.toBeNull();
+
+        expect(queryByDisplayValue('english name')).not.toBeNull();
+        expect(queryByDisplayValue('english description')).not.toBeNull();
+        expect(queryByDisplayValue('english nested field')).not.toBeNull();
+
+        expect(queryByDisplayValue('french name')).not.toBeNull();
+        expect(queryByDisplayValue('french description')).not.toBeNull();
+        expect(queryByDisplayValue('french nested field')).not.toBeNull();
+
+        fireEvent.change(getByLabelText('select locale'), {
+            target: { value: 'fr' },
+        });
+        expect(getByLabelText('en').getAttribute('hidden')).not.toBeNull();
+        expect(getByLabelText('fr').getAttribute('hidden')).toBeNull();
+    });
```

Comment:
I miss a test about validation and tab error status

Reply:
Added